### ISSUE TITLE
Require TypeScript 3.0 to build TSLint

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#4356]
**Is your feature request related to a problem? Please describe.**
In order to add TypeScript 3.0 unknown related rules, the TSLint needs to use TS 3.0 library.
For instance ts.SyntaxKind.UnknownKeyword identifier is not defined prior TypeScript 3.0 which is required for fixing #4355 .

**Describe the solution you'd like**
Update package.json to require TypeScript 3.0 and above.

**Describe alternatives you've considered**
If it is possible to use ts.SyntaxKind.UnknownKeyword without requiring TS 3.0 it would be better of course, but I don't know such way.

**Comment by JoshuaKGoldberg** [comment]
@NN--- this is an interesting discussion. Another workaround is from https://github.com/palantir/tslint/pull/4219/files#diff-8c6f0d081a57df6182dc05490ed38a63R166:

```typescript
        // Forwards-compatibility with TS3
        case (ts.SyntaxKind as any).UnknownKeyword:
```

...which should probably be changed to just `case ts.SyntaxKind.UnknownKeyword` after this goes in.

**Comment by JoshuaKGoldberg** [comment]
This'll be fixed by #4274

**Comment by NN---** [comment]
Seems like TS 3.0 requirement is a bug change and can be workarounded now.
